blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -59,64 +59,64 @@ All join provided by Kafka Streams were designed thinking with aggregation in mi
-- We create a `UUID` a startup for the worker
-- When a consumer connects to Kafka, we [listen to partitions](https://github.com/kestra-io/kestra/blob/4a61af45d668feab19313b9033826fa7075bf02b/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaWorkerTaskQueue.java#L157-L187) affected using a `ConsumerRebalanceListener` and we publish to Kafka a WorkerInstance with `UUID` and partition assigned.
-- For each task run, we publish a message with [TaskRunning](https://github.com/kestra-io/kestra/blob/4a61af45d668feab19313b9033826fa7075bf02b/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaWorkerTaskQueue.java#L112-L123) with the Worker UUID
+Thanks to Kafka consumer, we can know the specific partition that affected such a consumer. We use these valuable features to detect "dead" workers. How? Here is the logic:
+- We create a `UUID` startup for the worker.
+- When a consumer connects to Kafka, we [listen to affected partitions](https://github.com/kestra-io/kestra/blob/4a61af45d668feab19313b9033826fa7075bf02b/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaWorkerTaskQueue.java#L157-L187) using a `ConsumerRebalanceListener` and we publish to Kafka a WorkerInstance with `UUID` and partition assigned.
+- For each task run, we publish a message with [TaskRunning](https://github.com/kestra-io/kestra/blob/4a61af45d668feab19313b9033826fa7075bf02b/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaWorkerTaskQueue.java#L112-L123) with the Worker UUID.
 
 Now, let's handle the data stored in Kafka. The main logic is a [Kafka Stream](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/WorkerInstanceTransformer.java) that:
-- Create a global `KTable` with all the `WorkerInstance`
-- On every change, it will listen to `WorkerInstance` changed
-- If there is a new `WorkerInstance`, we look at the partition assigned on this one, if there is an overlap between this instance partitions and the previous, we are sure that the previous `WorkerInstance` is dead (in Kafka, you can't have 2 consumers on the same partition).
-- We only need to look at the Task affected to this `WorkerInstance` and resend them for processing.
+- Creates a global `KTable` with all the `WorkerInstance.`
+- Listens to `WorkerInstance` changed for every change.
+- If there is a new `WorkerInstance,` we look at the partition assigned. Where there is an overlap between this instance's partitions and the previous ones, we are sure that the previous `WorkerInstance` is dead. This is because you can't have two consumers on the same partition in Kafka. 
+- We only need to look at the task affected for this `WorkerInstance` and resend it for processing.
 
-Et voila 🎉 We have detection of dead consumers for free only with Kafka  API.
+Et voila 🎉 We can detect dead consumers for free with Kafka API.
 
 ## State store `all()`
```

Comment:
Again, you may want to consider including a sentence or two here providing context to the subheader and connecting it to the rest of the article.